core/src/main/java/io/grpc/internal/TransportSet.java

```diff
@@ -420,11 +424,8 @@ public void transportShutdown(Status s) {
 
             // Initiate backoff
             // Transition to TRANSIENT_FAILURE
-            DelayedClientTransport newDelayedTransport = new DelayedClientTransport(appExecutor);
-            transports.add(newDelayedTransport);
-            newDelayedTransport.start(new BaseTransportListener(newDelayedTransport));
-            activeTransport = newDelayedTransport;
-            scheduleBackoff(newDelayedTransport);
+            assert !delayedTransport.isInBackoffPeriod();
```

Comment:
Not sure about using assert vs. Precondtions.checkState, but we've been exclusively using Preconditions throughout the codebase, so may be better to stay consistent.